docs/tls-authentication.md

```diff
@@ -0,0 +1,248 @@
+## Generating Certificates
+
+> **NOTE**: If you're setting up backend nodes and a frontend wallet on a same machine, 
+> you can ignore the _personalized_ installation and fully rely on the _default_ case.
+
+The following steps are described as if executed from a _Unix_ system (Linux or Mac OS) however,
+scripts used are also available on _Windows_ and steps could be easily transposed to this
+system.
+
+### Default (localhost, 3 years)
+
+#### Downloading Script
+
+ _[daedalus](https://github.com/input-output-hk/daedalus)_ provides a script
+ `installers/build-certificates-unix.sh` (or a windows equivalent
+ `installers/build-certificates-win64.bat`) to generate default certificates for both
+ _cardano-sl_ and _daedalus_, as well as a CA public certificate used to sign the formers. 
+
+Make sure you have an Internet access and download the installer script and the necessary
+configuration files as such:
+
+```
+$ wget -q https://raw.githubusercontent.com/input-output-hk/daedalus/master/installers/build-certificates-unix.sh
+$ wget -q https://raw.githubusercontent.com/input-output-hk/daedalus/master/installers/{ca,server,client}.conf
+```
+
+#### Running Script
+
+Run the script to generate the certificates as follows:
+
+```
+$ bash ./build-certificates-unix.sh 
+```
+
+Once done (`Oll Korrect`), a folder `tls` should have been created in your current directory
+containing, at least, the following files:
+
+```
+--| tls
+-----| ca
+--------| ca.crt
+-----| client
+--------| client.crt
+--------| client.key
+-----| server
+--------| server.crt
+--------| server.key
+```
+
+Those are the certificates and private keys that _daedalus_ and _cardano-sl_ will require to
+talk to each other. See [Using Certificates](#using-certificates) for more details.
+
+
+### Personalized 
+
+Follow the steps described in [Default | Downloading Script](#downloading-script) to first pull
+the build script and the required configuration files.
+
+> **NOTE:** You can also simply use compatible certificates that have been signed by an
+> existing authority. Make sure you have the public certificate of that authority in your 
+> possession.
+
+#### Custom Server Domains
+
+The `build-certificates-*` script will generate server's certificates for `localhost`. This
+seemingly assume several things:
+
+  - The backend nodes and frontend are running on the same machine
+  - You don't need to enforce the backend domain (multiple services may be running on `localhost`)
+
+In any case, you might need to run the backend on a different domain and therefore, identify it
+using something else than `localhost`. To do so, modify the `[ server_alt_names ]` section in
+the `ca.conf` downloaded earlier:
+
+```
+[ server_alt_names ]
+DNS.1                   = localhost.localdomain
+DNS.2                   = localhost
+DNS.3                   = 127.0.0.1
+DNS.4                   = ::1
+```
+
+and replace DNS references with your actual domain. For instance:
+
+```
+[ server_alt_names ]
```

Comment:
:) ... We kinda had to, because using only the common name results in a very strict name validation and hosts like "127.0.0.1" are treated differently from "localhost" (which is not _wrong_ so to speak, but doesn't properly work for our case). It's more a commodity, but at least it now gives room for future user to easily tweak that, in a right way.